### PR TITLE
Mesh upstream discovery

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/supergloo/issues/379

### CODE REVIEW

File: api/v1/mesh.proto [hunk 1]
```diff
@@ -63,8 +63,11 @@ message DiscoveryMetadata {
     // namespace which the mesh is installed into
     string installation_namespace = 4;
 
+    // upstreams which point to injected pods in the mesh
+    repeated core.solo.io.ResourceRef mesh_upstreams = 5;
```
Comment (ilackarms) [inline api/v1/mesh.proto]: shouldn't this go in the discovery metadata?

Comment (EItanya) [reply api/v1/mesh.proto]: Good point

File: pkg/api/custom/clients/kubernetes/pod_base_client.go [hunk 2]
```diff
@@ -17,19 +17,21 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-type ResourceClient struct {
+type KubeCoreResourceClient struct {
```
Comment (ilackarms) [inline pkg/api/custom/clients/kubernetes/pod_base_client.go]: extract this to a common file

File: pkg/meshdiscovery/config/istio/syncer.go [hunk 3]
```diff
@@ -4,19 +4,33 @@ import (
...
+	"github.com/solo-io/supergloo/pkg/meshdiscovery/clientset"
+	"github.com/solo-io/supergloo/pkg/meshdiscovery/discovery/istio"
+	"github.com/solo-io/supergloo/pkg/meshdiscovery/utils"
 	"go.uber.org/zap"
 )
 
 const (
-	injectionLabel = "istio-injection"
+	namespaceInjectionLabel = "istio-injection"
+	enabled                 = "true"
```
Comment (ilackarms) [inline pkg/meshdiscovery/config/istio/syncer.go]: enabled

**Pull request merged**